query-node/mappings/giza/storage/index.ts

```diff
@@ -0,0 +1,636 @@
+      id,
+      owner: getStaticBagOwner(bagId),
+    })
+    await store.save<StorageBag>(newBag)
+    return newBag
+  }
+  return bag
+}
+
+async function getBag(store: DatabaseManager, bagId: BagId, relations?: 'objects'[]): Promise<StorageBag> {
+  return bagId.isStatic
+    ? getStaticBag(store, bagId.asStatic, relations)
+    : getDynamicBag(store, bagId.asDynamic, relations)
+}
+
+async function getDistributionBucketOperatorWithMetadata(
+  store: DatabaseManager,
+  id: string
+): Promise<DistributionBucketOperator> {
+  const operator = await store.get(DistributionBucketOperator, {
+    where: { id },
+    relations: ['metadata', 'metadata.nodeLocation', 'metadata.nodeLocation.coordinates'],
+  })
+  if (!operator) {
+    throw new Error(`DistributionBucketOperator not found by id: ${id}`)
+  }
+  return operator
+}
+
+async function getStorageBucketWithOperatorMetadata(store: DatabaseManager, id: string): Promise<StorageBucket> {
+  const bucket = await store.get(StorageBucket, {
+    where: { id },
+    relations: ['operatorMetadata', 'operatorMetadata.nodeLocation', 'operatorMetadata.nodeLocation.coordinates'],
+  })
+  if (!bucket) {
+    throw new Error(`StorageBucket not found by id: ${id}`)
+  }
+  return bucket
+}
+
+async function getDistributionBucketFamilyWithMetadata(
+  store: DatabaseManager,
+  id: string
+): Promise<DistributionBucketFamily> {
+  const family = await store.get(DistributionBucketFamily, {
+    where: { id },
+    relations: ['metadata', 'metadata.boundary'],
+  })
+  if (!family) {
+    throw new Error(`DistributionBucketFamily not found by id: ${id}`)
+  }
+  return family
+}
+
+// BUCKETS
+
+export async function storage_StorageBucketCreated({ event, store }: EventContext & StoreContext): Promise<void> {
+  const [
+    bucketId,
+    invitedWorkerId,
+    acceptingNewBags,
+    dataObjectSizeLimit,
+    dataObjectCountLimit,
+  ] = new Storage.StorageBucketCreatedEvent(event).params
+
+  const storageBucket = new StorageBucket({
+    id: bucketId.toString(),
+    acceptingNewBags: acceptingNewBags.isTrue,
+    dataObjectCountLimit: new BN(dataObjectCountLimit.toString()),
+    dataObjectsSizeLimit: new BN(dataObjectSizeLimit.toString()),
+  })
+  if (invitedWorkerId.isSome) {
+    const operatorStatus = new StorageBucketOperatorStatusInvited()
+    operatorStatus.workerId = invitedWorkerId.unwrap().toNumber()
+    storageBucket.operatorStatus = operatorStatus
+  } else {
+    storageBucket.operatorStatus = new StorageBucketOperatorStatusMissing()
+  }
+  await store.save<StorageBucket>(storageBucket)
+}
+
+export async function storage_StorageOperatorMetadataSet({ event, store }: EventContext & StoreContext): Promise<void> {
+  const [bucketId, , metadataBytes] = new Storage.StorageOperatorMetadataSetEvent(event).params
+  const storageBucket = await getStorageBucketWithOperatorMetadata(store, bucketId.toString())
+  storageBucket.operatorMetadata = await processStorageOperatorMetadata(
```

Comment:
Why don't we throw an error here on invalid metadata similar to other invalid data?

Reply:
The errors are only thrown on some unexpected state which should never occur if the node is working correctly.
Invalid metadata is a different case, because the runtime does not check its validity, so it's a _normal_ condition that the metadata is invalid. We're just logging this and performing no action in this case.